haystack/nodes/answer_generator/openai.py

```diff
@@ -199,6 +199,14 @@ def predict(
                 )
             raise openai_error
 
+        for ans in res["choices"]:
+            if ans["finish_reason"] == "length":
+                logger.warning(
+                    "At least one of the answers has been truncated before reaching a natural stopping point."
+                    "Consider increasing max_tokens parameter to produce better answers."
+                )
+                break
```

Comment:
Perhaps instead of breaking after the first detection of `length` we could count how many answers were truncated and relay that information to the user? And the error message could be something like 
```python
logger.warning(
    "%s out of the %s answers have been truncated before reaching a natural stopping point."
    "Consider increasing the max_tokens parameter to allow for longer answers.",
    n_trunc_ans, top_k)
```